src/pyscheduling/SMSP/interface.py

```diff
@@ -93,10 +93,12 @@ def add_constraints(self, constraints):
         """
         if type(constraints) == list : 
             for constraint in constraints :
-                if type(constraint) != Constraints : raise  TypeError("Only Constraints Enum elements are allowed :\n"+Constraints.toString())
+                print(constraint,Constraints)
+                print(issubclass(constraint,Constraints))
+                if type(constraint) != Constraints : raise  TypeError("Only Constraints Enum elements are allowed :\n"+Constraints_Enum.to_string())
             self.constraints.extend([constraint for constraint in constraints if constraint not in self.constraints])
         elif (type(constraints) == Constraints) and (constraints not in self.constraints): self.constraints.append(constraints)
-        else: raise  TypeError("Only Constraints Enum elements are allowed :\n"+Constraints.toString())
+        else: raise  TypeError("Only Constraints Enum elements are allowed :\n"+Constraints_Enum.to_string())
```

Comment:
I think it's Constraints in here and not Constraints_Enum, your list of constraints is of Class Constraints